bluemira/magnetostatics/circuits.py

```diff
@@ -58,22 +58,44 @@ def _generate_sources(
         self.d_l = np.diff(self.shape, axis=0)
         self.midpoints = self.shape[:-1, :] + 0.5 * self.d_l
         sources = []
+        warning_called = False
         for midpoint, d_l, beta, alpha in zip(
             self.midpoints, self.d_l, betas, alphas, strict=False
         ):
             d_l_norm = d_l / np.linalg.norm(d_l)
             t_vec = np.cross(d_l_norm, normal)
+            if source_class is PolyhedralPrismCurrentSource:
+                source = source_class(
+                    midpoint,
+                    d_l,
+                    normal,
+                    t_vec,
+                    *xs_args,
+                    alpha=alpha,
+                    beta=beta,
+                    current=current,
+                    bypass_endcap_error=True,
```

Comment:
Not sure this is the right idea as it means if you use polyhedralprism for circuits it will turn off even if you don't want to. Admittedly this will not be for many case (ring or perfectly discretised rectangle) but should I add an extra qualifier to this class to handle this?

Reply:
I think as long it warns its ok, but we just need to make people aware that the polyhedralprism has some sharp edges.